pkg/service/egress.go

```diff
@@ -228,16 +254,30 @@ func (s *EgressService) UpdateStream(ctx context.Context, req *livekit.UpdateStr
 	if err := EnsureRecordPermission(ctx); err != nil {
 		return nil, twirpAuthError(err)
 	}
-	if s.rpcClient == nil {
-		return nil, ErrEgressNotConnected
+
+	if !s.useNewRPC {
+		s.useNewRPC = useNewEgressRPC(s.es)
 	}
 
-	info, err := s.rpcClient.SendRequest(ctx, &livekit.EgressRequest{
-		EgressId: req.EgressId,
-		Request: &livekit.EgressRequest_UpdateStream{
-			UpdateStream: req,
-		},
-	})
+	var info *livekit.EgressInfo
+	var err error
+	if s.useNewRPC {
```

Comment:
For this to work if we flip the switch mid active egress, I suppose we'll have egress servers listen on both RPC channels as well?

Reply:
yeah